service/k8sclient/k8sclient.go

```diff
@@ -827,17 +815,32 @@ func (c *K8sClient) getPerconaXtraDBClusters(ctx context.Context) ([]XtraDBClust
 	return res, nil
 }
 
-func getPXCState(status *pxc.PerconaXtraDBClusterStatus) ClusterState {
-	if status == nil {
+func (c *K8sClient) getPXCState(ctx context.Context, cluster *pxc.PerconaXtraDBCluster, crAndPodsMatchFunc func(context.Context, common.DatabaseCluster) (bool, error)) ClusterState {
+	if cluster == nil || cluster.Status == nil {
 		return ClusterStateInvalid
 	}
-	clusterState, ok := pxcStatesMap[status.Status]
+	// Handle paused state for operator version >= 1.9.0 and for operator version <= 1.8.0.
+	if cluster.Status.Status == pxc.AppStatePaused || (cluster.Spec.Pause && cluster.Status.Status == pxc.AppStateReady) {
```

Comment:
Or do you mean `cluster.Status.Status`? @oter I don't know why it's `Status` and not `State`. It's been like that since I joined. Maybe @BupycHuk can tell what was the reason.